drivers/fuel_gauge/ltc2959/ltc2959.c

```diff
@@ -0,0 +1,544 @@
+			ret = ltc2959_read16(dev, LTC2959_REG_CURRENT_MSB, &raw_current);
+			if (ret < 0) {
+				return ret;
+			}
+			/* Signed 16-bit value from ADC */
+			int16_t current_raw = (int16_t) raw_current;
+			val->current = current_raw * cfg->current_lsb_uv;
+			break;
+
+		case FUEL_GAUGE_TEMPERATURE:
+			uint16_t raw_temp;
+			ret = ltc2959_read16(dev, LTC2959_REG_TEMP_MSB, &raw_temp);
+			if (ret < 0) {
+				return ret;
+			}
+			/**
+			 * NOTE:
+			 * Temp is in deciKelvin as per API requirements.
+			 * from the datasheet:
+			 * T(°C) = 825 * (raw / 65536) - 273.15
+			 * T(dK) = 8250 * (raw / 65536)
+			 * 65536 = 2 ^ 16, so we can avoid division altogether.
+			 */
+			val->temperature = ((uint32_t) raw_temp * LTC2959_TEMP_K_SF) >> 16;
+			break;
+
+		case FUEL_GAUGE_CHARGE_CURRENT:
+			uint32_t raw;
+			ret = ltc2959_read32(dev, LTC2959_REG_ACC_CHARGE_3, &raw);
+			if (ret < 0) {
+				return ret;
+			}
+			val->chg_current = ((uint64_t) raw * LTC2959_CHARGE_UAH_SF) >> 16;
+			break;
+
+		case FUEL_GAUGE_ADC_MODE:
+			ret = ltc2959_get_adc_mode(dev, (uint8_t*) val->flags);
+			break;
+
+		case FUEL_GAUGE_HIGH_VOLTAGE_ALARM:
+			ret = ltc2959_get_voltage_threshold_uv(
+			        dev, true, (uint32_t*) &val->voltage);
+			break;
+
+		case FUEL_GAUGE_LOW_VOLTAGE_ALARM:
+			ret = ltc2959_get_voltage_threshold_uv(
+			        dev, false, (uint32_t*) &val->voltage);
+			break;
+
+		case FUEL_GAUGE_HIGH_CURRENT_ALARM:
+			ret = ltc2959_get_current_threshold_ua(dev, true, (int32_t*) &val->current);
+			break;
+
+		case FUEL_GAUGE_LOW_CURRENT_ALARM:
+			ret = ltc2959_get_current_threshold_ua(
+			        dev, false, (int32_t*) &val->current);
+			break;
+
+		case FUEL_GAUGE_HIGH_TEMPERATURE_ALARM:
+			ret = ltc2959_get_temp_threshold_dK(dev, true, &val->temperature);
+			break;
+
+		case FUEL_GAUGE_LOW_TEMPERATURE_ALARM:
+			ret = ltc2959_get_temp_threshold_dK(dev, false, &val->temperature);
+			break;
+
+		case FUEL_GAUGE_GPIO_VOLTAGE_UV:
+			ret = ltc2959_get_gpio_voltage_uv(dev, &val->voltage);
+			break;
+
+		case FUEL_GAUGE_HIGH_GPIO_ALARM:
+			int32_t hi_gpio;
+			ret = ltc2959_get_gpio_threshold_uv(dev, true, &hi_gpio);
+			if (ret == 0) {
+				val->voltage = hi_gpio;
+			}
+			break;
+
+		case FUEL_GAUGE_LOW_GPIO_ALARM:
+			int32_t lo_gpio;
+			ret = ltc2959_get_gpio_threshold_uv(dev, false, &lo_gpio);
+			if (ret == 0) {
+				val->voltage = lo_gpio;
+			}
+			break;
+
+		case FUEL_GAUGE_CC_CONFIG:
+			uint8_t cc_ctrl;
+			ret = ltc2959_get_cc_config(dev, &cc_ctrl);
+			if (ret == 0) {
+				val->fg_status = cc_ctrl;
+			}
+			break;
+
+		default:
+			return -ENOTSUP;
+	}
+	return ret;
+}
+
+static int ltc2959_set_prop(
+        const struct device* dev, fuel_gauge_prop_t prop, union fuel_gauge_prop_val val)
+{
+	int ret = 0;
+
+	switch (prop) {
+		case FUEL_GAUGE_ADC_MODE:
+			ret = ltc2959_set_adc_mode(dev, val.fg_status);
+			break;
+
+		case FUEL_GAUGE_LOW_VOLTAGE_ALARM:
+			ret = ltc2959_set_voltage_threshold_uv(dev, false, val.voltage);
```

Comment:
Wrong `fuel_gauge_prop_val`
```suggestion
			ret = ltc2959_set_voltage_threshold_uv(dev, false, val.low_voltage_alarm);
```